File: api/filing/models/tds_model.py
"""TDS (Tax Deducted at Source) model."""
from .api_base_model import ApiBaseModel


class TDSModel(ApiBaseModel):
    tds_id: int | None = None
    filing_id: int | None = None
    deductor_name: str | None = None
    tan: str | None = None
    pan: str | None = None
    income_source: str | None = None
    tds_section: str | None = None
    amount_paid: float | str | int | None = None
    tax_deducted: float | str | int | None = None
    tds_certificate_number: str | None = None
    quarter: str | None = None

File: web/filing/models/tax-credits/tds-model.ts
/**
 * TDS Model
 * Tax Deducted at Source
 */
export interface TDSModel {
  tdsId?: number | null
  filingId: number
  deductorName: string
  tan: string
  pan:string |null
  incomeSource?: string | null
  tdsSection?: string | null
  amountPaid?: number | null
  taxDeducted?: number | null
  tdsCertificateNumber?: string | null
  quarter?: string | null
}
export interface TDSEntryWidgetProps {
  isOpen: boolean
  onClose: () => void
  filingId: number | null
  entityId?: number | string
  initialData?: Record<string, any> | null
  onSuccess?: (action: 'created' | 'updated', data: any) => void
  errorField?: string | null
  errorMessage?: string | null  // Error message to display for the field
  fieldErrors?: Record<string, string> | null  // Multiple field errors: { fieldName: errorMessage }
}


export interface TDSResponse {
  data: TDSModel
  message?: string
}

export interface TDSListResponse {
  data: TDSModel[]
  message?: string
}